core/src/main/resources/jenkins/model/Jenkins/login.jelly

```diff
@@ -23,58 +23,119 @@ THE SOFTWARE.
-            <tr>
-              <td>${%Password}:</td>
-              <td><input type="password" name="j_password" /></td>
-            </tr>
-            <j:if test="${app.security.name()=='SECURED' and !app.disableRememberMe}">
-              <!-- remember me service only available for Acegi -->
-              <tr>
-                <td align="right"><input id="remember_me" type="checkbox" name="remember_me" /></td>
-                <td><label for="remember_me">${%Remember me on this computer}</label></td>
-              </tr>
-            </j:if>
-          </table>
-          <input type="hidden" name="from" value="${request.getParameter('from')}" />
-          <f:submit value="${%login}" />
-          <script>
-            $('j_username').focus();
-          </script>
-        </form>
+<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form" xmlns:x="jelly:xml" xmlns:st="jelly:stapler">
+  <!-- deactivate all caching -->
+  <st:setHeader name="Expires" value="0" />
+  <st:setHeader name="Cache-Control" value="no-cache,no-store,must-revalidate" />
+  <j:new var="h" className="hudson.Functions" /><!-- instead of JSP functions -->
+  <!-- needed for cli -->
+  <j:set var="extensionsAvailable" value="${h.extensionsAvailable}"/>
+  <j:if test="${request.servletPath=='/' || request.servletPath==''}">
+    ${h.advertiseHeaders(response)}
+    <j:if test="${extensionsAvailable}">
+      <j:forEach var="pd" items="${h.pageDecorators}">
+        <st:include it="${pd}" page="httpHeaders.jelly" optional="true"/>
+      </j:forEach>
+    </j:if>
+  </j:if>
+  <!-- response contentType header -->
+  <st:contentType value="text/html;charset=UTF-8" />
+  <!--
+    needed to generate a session if non exists,
+    without it we would add ";jsessionid=" to the url which will result in a 404
+  -->
+  <j:set var="_" value="${request.getSession()}"/>
+  <!-- get default/commong page variable -->
+  ${h.initPageVariables(context)}
+  <x:doctype name="html" />
+  <!-- needed for the redirect after login -->
+  <j:set var="from" value="${error ? request.session.getAttribute('from') : request.getParameter('from')}"/>
+  <!-- in case of error we want to surround the form elements with an error hint -->
+  <j:set var="inputClass" value="${error ? 'danger' : 'normal'}"/>
+  <j:set var="loginDecorator" value="${h.loginPageDecorator}"/>
+  <!-- real deal starting here -->
+  <html>
+    <head data-rooturl="${rootURL}" data-resurl="${resURL}" resURL="${resURL}">
+      <title>${h.appendIfNotNull('Login', ' [Jenkins]', 'Jenkins')}</title>
+      <!-- we do not want bots on this page -->
+      <meta name="ROBOTS" content="NOFOLLOW" />
+      <!-- css styling, will fallback to default implementation -->
+      <st:include it="${loginDecorator}" page="login-css.jelly" optional="true"/>
+    </head>
+    <body>
+      <j:choose>
+        <j:when test="${it.setupWizard.isUsingSecurityToken()}">
+          <st:include it="${it.setupWizard}" page="authenticate-security-token"/>
```

Comment:
Did you check the setup wizard page is not affected by the _new default_ login css?